File: objectify-autoconfiguration/src/main/java/com/yoloo/server/objectify/translators/CuckooFilterTranslatorFactory.java
package com.yoloo.server.objectify.translators;

import com.google.cloud.datastore.Blob;
import com.google.cloud.datastore.BlobValue;
import com.google.cloud.datastore.Value;
import com.google.cloud.datastore.ValueType;
import com.googlecode.objectify.impl.translate.SimpleTranslatorFactory;
import net.cinnom.nanocuckoo.NanoCuckooFilter;
import org.springframework.util.SerializationUtils;

import java.util.Objects;

class CuckooFilterTranslatorFactory extends SimpleTranslatorFactory<NanoCuckooFilter, Blob> {

  CuckooFilterTranslatorFactory() {
    super(NanoCuckooFilter.class, ValueType.BLOB);
  }

  @Override
  protected NanoCuckooFilter toPojo(Value<Blob> value) {
    return (NanoCuckooFilter) SerializationUtils.deserialize(value.get().toByteArray());
  }

  @Override
  protected Value<Blob> toDatastore(NanoCuckooFilter value) {
    byte[] bytes = SerializationUtils.serialize(value);
    return BlobValue.of(Blob.copyFrom(Objects.requireNonNull(bytes)));
  }
}

File: service.user/src/main/kotlin/com/yoloo/server/user/infrastructure/objectify/translators/CuckooFilterTranslatorFactory.kt
package com.yoloo.server.user.infrastructure.objectify.translators

import com.google.appengine.api.datastore.Blob
import com.yoloo.server.objectify.translators.AbstractSimpleTranslatorFactory
import net.cinnom.nanocuckoo.NanoCuckooFilter
import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream
import java.io.ObjectInputStream
import java.io.ObjectOutputStream

class CuckooFilterTranslatorFactory :
    AbstractSimpleTranslatorFactory<NanoCuckooFilter, Blob>(NanoCuckooFilter::class.java) {
    override fun createTranslator(): SimpleTranslator<NanoCuckooFilter, Blob> {
        return object : SimpleTranslator<NanoCuckooFilter, Blob> {
            override fun loadValue(datastoreValue: Blob): NanoCuckooFilter {
                val bytes = datastoreValue.bytes
                val byteInputStream = ByteArrayInputStream(bytes)
                val objectInputStream = ObjectInputStream(byteInputStream)
                return objectInputStream.readObject() as NanoCuckooFilter
            }

            override fun saveValue(pojoValue: NanoCuckooFilter?): Blob {
                val byteOutputStream = ByteArrayOutputStream()
                val objectOutputStream = ObjectOutputStream(byteOutputStream)
                objectOutputStream.writeObject(pojoValue)
                val bytes = byteOutputStream.toByteArray()
                return Blob(bytes)
            }

        }
    }
}